datafusion/common/src/config.rs

```diff
@@ -1566,8 +1517,68 @@ impl TableOptions {
     /// # Parameters
     ///
     /// * `format`: The file format to use (e.g., CSV, Parquet).
-    pub fn set_config_format(&mut self, format: ConfigFileType) {
-        self.current_format = Some(format);
+    pub fn get_table_format_options(&self, format: ConfigFileType) -> TableFormatOptions {
+        match format {
+            ConfigFileType::CSV => {
+                let options = self.csv.clone();
+                TableFormatOptions::Csv {
+                    options,
+                    extensions: self.extensions.clone(),
+                }
+            }
+            #[cfg(feature = "parquet")]
+            ConfigFileType::PARQUET => {
+                let options = self.parquet.clone();
+                TableFormatOptions::Parquet {
+                    options,
+                    extensions: self.extensions.clone(),
+                }
+            }
+            ConfigFileType::JSON => {
+                let options = self.json.clone();
+                TableFormatOptions::Json {
+                    options,
+                    extensions: self.extensions.clone(),
+                }
+            }
+        }
+    }
+
+    /// Initializes a new `TableOptions` from a hash map of string settings.
+    ///
+    /// # Parameters
+    ///
+    /// * `settings`: A hash map where each key-value pair represents a configuration setting.
+    ///
+    /// # Returns
+    ///
+    /// A result containing the new `TableOptions` instance or an error if any setting could not be applied.
+    pub fn from_string_hash_map(settings: &HashMap<String, String>) -> Result<Self> {
+        let mut ret = Self::default();
+        for (k, v) in settings {
+            ret.set(k, v)?;
+        }
+
+        Ok(ret)
+    }
+
+    /// Modifies the current `TableOptions` instance with settings from a hash map.
+    ///
+    /// # Parameters
+    ///
+    /// * `settings`: A hash map where each key-value pair represents a configuration setting.
+    ///
+    /// # Returns
+    ///
+    /// A result indicating success or failure in applying the settings.
+    pub fn alter_with_string_hash_map(
```

Comment:
maybe `update` would be clearer as `alter` might imply some change other than updating the map